### DISCUSSION TITLE
Chips.setBorderColor(Color) also set the hover/pushed?

### DISCUSSION THREAD

**Discussion by laeubi** [EclipseNebula/nebula#468] [Ideas]
I noticed an (unexpected) behavior when using the [Chips](https://wiki.eclipse.org/Nebula_Chips) widget.

If I create a "push" widget (by the way according to the description `SWT.TOGGLE` would be more suitable here than `SWT.PUSH`)
and call setBorderColor(Color) I end up with no border at all.

This is, because in this case  the `hoverBorderColor` and `pushedStateBorderColor` is used.

I'd like to discuss if we probably set all three colors in case one calls setBorderColor() instead as this is similar to css where one often often do e.g. `border-color: red; border-color-left: blue;` to set all sides of the border to red and then only adjust the one that should be different.

**Comment by lcaron** [comment]
Hi Christoph,

That's interesting, you're right.

Do you want to work on this topic ? Feel free to ask.

Take care,

Laurent

**Reply by laeubi** [reply]
> Do you want to work on this topic ?

Probably yes, but first wanted to get some feedback if there are probably side-effects or a special consideration why it works this way.
